docs/public/secure-registry.adoc

```diff
@@ -0,0 +1,35 @@
+== Introduction to secure registry
+
+=== What is a secure registry?
+
+A secure registry is a registry that user can only use credential to access.
+
+=== Where to host secure registry?
+
+User can host secure registry on private GitHub repository, Enterprise GtiHub repository and cluster.
+
+== Steps for adding secure registry when registry is hosted on GitHub repository
+
+=== Step 1: Creating a new GitHub repository to host the registry
+
+Please follow https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-new-repository[this instruction] to create a new GitHub repository and push the registry to the repository. The sample GitHub-hosted registry can be found https://github.com/odo-devfiles/registry/[here].
+
+=== Step 2: Creating a personal access token 
+
+Please follow https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token[this instruction] to create a personal access token, select `repo` as token scope.
+
+=== Step 3: Keyring setup
+
+There is no specific keyring setup for secure registry, you only need to ensure the keyring is working properly on your system, if you hit issues please follow the following instructions to troubleshoot the issues of your keyring with respective to the corresponding platforms.
```

Comment:
I think we should provide some proper instructions on setting up the keyring on Linux. When I had to do this with Codewind on my Linux box, it took a fair bit of googling.

Reply:
Done